client/tools/devServer.js

```diff
@@ -0,0 +1,106 @@
+// This file configures the development web server
+// which supports hot reloading and synchronized testing.
+
+// Require Browsersync along with webpack and middleware for it
+import browserSync from 'browser-sync';
+// Required for react-router browserHistory
+// see https://github.com/BrowserSync/browser-sync/issues/204#issuecomment-102623643
+import history from 'connect-history-api-fallback';
+import webpack from 'webpack';
+import webpackDevMiddleware from 'webpack-dev-middleware';
+import webpackHotMiddleware from 'webpack-hot-middleware';
+import proxy from 'http-proxy-middleware';
+import { chalkProcessing } from './chalkConfig';
+
+import config from '../webpack.config.dev';
+
+const proxyTargets = [
+  {
+    source: '/api/**',
+    target: 'http://backend:3000',
+  },
+  {
+    source: '/env',
+    target: 'http://backend:3000',
+  },
+  {
+    source: '/healthz',
+    target: 'http://backend:3000',
+  },
+  {
+    source: '/maps/**',
+    target: 'http://windshaft:4000',
+    pathRewrite: {
+      '/maps': '',
+    },
+  },
+  {
+    source: '/share/**',
+    target: 'http://backend:3000',
+  },
+  {
+    source: '/verify/**',
+    target: 'http://backend:3000',
+  },
+];
+const bundler = webpack(config);
+
+console.log(chalkProcessing('Starting dev server...'));
+
+// Run Browsersync and use middleware for Hot Module Replacement
+browserSync({
+  port: 3030,
+  ui: false,
+  open: false,
+  server: {
+    baseDir: 'src',
+
+    middleware: [
+
+      history({
+        rewrites: [
+          { from: '^/favicon.ico$', to: 'favicon.ico' },
+          { from: '^/s/.*$', to: '/index-pub.html' },
+          // { from: '^.*$', to: '/index.html' },
```

Comment:
this rewrite was problematic because it was making requests for js files getting returned the index.html file - we just need to check the same thing is happening on the index-pub.html page - Im unfamiliar with this @gabemart

Reply:
The index-pub page is served for shared visualisations and dashboards, and you're suspicion is correct, it's not working currently in the dev environment.

I think we can remove the config object entirely, as the favicon request is also not working. Removing the object (i.e. just `history()`) restores the function of `index-pub.html` for shared visualisations in the dev env

Reply:
is it ok now?